src/main/scala/beam/agentsim/infrastructure/ChargingNetwork.scala

```diff
@@ -37,11 +40,17 @@ class ChargingNetwork(managerId: Id[VehicleManager], chargingStationsQTree: Quad
   def vehicles: Map[Id[BeamVehicle], ChargingVehicle] = chargingStationMap.flatMap(_._2.vehicles)
 
   /**
-    * lookup a station from a parking stall
-    * @param stall the parking stall
-    * @return charging station
+    * lookup a station from attributes
+    * @param tazId the taz id
+    * @param parkingType the parking type
+    * @param chargingPointType the charging type
+    * @return
     */
-  def lookupStation(stall: ParkingStall): Option[ChargingStation] = chargingStationMap.get(ChargingZone.to(stall))
+  def lookupStation(
+    tazId: Id[TAZ],
+    parkingType: ParkingType,
+    chargingPointType: ChargingPointType
+  ): Option[ChargingStation] = chargingStationMap.get(s"cs_${managerId}_${tazId}_${parkingType}_${chargingPointType}")
```

Comment:
it seems there is a counterpart of the code which fills the map with these keys. 
If it is possible, it would be better to move key generation to one function and reuse this function in two (or more) places

Reply:
Done!